File: in_Python/0387 First Unique Character in a String.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        dict = {}
        for i in range(len(s)):
            c = s[i]
            if c not in dict:
                dict[c] = 1
            else:
                dict[c] += 1
        if 1 not in dict.values(): return -1
        
        res = len(s)+1
        for k in dict:
            if dict[k] == 1:
                v = s.index(k)
                if v < res:
                    res = v
        return res
                

File: in_Java/0387 First Unique Character in a String.java
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character,Integer> hm = new HashMap<>();
        for(char c : s.toCharArray()){
            if(!hm.containsKey(c)){
                hm.put(c,1);
            }else{
                hm.put(c,hm.get(c)+1);
            }
        }
        int res = -1;
        for(int i = 0; i < s.length(); i++){
            if(hm.get(s.charAt(i)) == 1){
                res = i;
                break;
            }
        }
        return res;
    }
}